src/writers/utils.ts

```diff
@@ -65,3 +66,9 @@ export const buildUtilsContent = (
 
   return lines.join(`\n`);
 };
+
+export const isOptionalField = (f: Field) => !f.required;
+
+export const isOptionalType = (t: VisitedType) =>
+  isInterface(t.class) && t.class.fields.some(isOptionalField);
```

Comment:
Can these two go into `predicates.ts` instead with all the other `(t: T) => boolean`s?

Also, the naming here isn't quite perfect: `s/isOptionalType/hasOptionalFields/` would be a smidge better.